### DISCUSSION TITLE
Slow App Performance

### DISCUSSION THREAD

**Discussion by gacolitti** [ghostfolio/ghostfolio#3268] [General]
I’m trying to understand why it takes so long to load pages in the app. Most of the time it takes 30+ seconds to load certain pages like Portfolio > Analysis, and others take so long to load I just give up. 

I’m guessing it is trying to fetch data each time the page loads. Perhaps there is a way to cache this data, set a refresh schedule in the app, or otherwise optimize the performance so the app is more responsive. 

For reference, I have 11 accounts and 984 activities. I’m running v2.71.0 self-hosted locally.

**Comment by chrisleavoy** [comment]
Overall this app seems pretty well put together, but yeah, the page load performance is poor. Even with just 1 account and 1 activity its 5+ seconds to load. Similar behavior on the premium hosted version. This was the only issue I found on the subject so far. Is there some docs or pointers around what the architecture is that results in slow page loads? I'm curious about hacking on this project.

**Reply by dtslvr** [reply]
Thank you for your feedback, @chrisleavoy.

Over the last few weeks, we have been working on the optimization in this area and have started making preparations. The aim is to centralize all calculations in the `PortfolioCalculator` in order to cache the portfolio snapshot (internal representation of all activities).

It would be great if you could take a closer look at the code and get back with specific ideas for improvement.

**Comment by MotafokaBR** [comment]
The changes on May 2nd and May the 4th (be with you) seem to be in the right direction, as they made the overall performance on my self hosted Ghostfolio way better than before.

**Reply by karter16** [reply]
Yep same here - subjectively I'm seeing much much better performance in the last couple of versions.

**Reply by zaromarco** [reply]
in my case the overall performance are decreased to a point of no usability of the page portfolio-analysis.
i'm using the latest version of May and docker container is set to use a maximum of 4gb of ram with medium cpu priority.
i don't know the min specs to use ghostfolio in docker
i have near 450 activities and many time also activity page freeze for many many seconds

**Comment by Dilosen** [comment]
HOLY SMOKES is this app running slow. AS soon as I import some transactions (~25 ) it just freezes. I've done umbrel oS, standalone and tried their website and ALL are ridiculously slow. Sometimes even just changing the currency will make it freeze up. For the life of me, I don't know why it's like this.

In fact it crashed/slowed the umbrel OS down I had to reinstall it. Prior to not installing Ghostfolio, everything ran relatively smoothly on the Umbrel OS, then as soon as I added the app, it just stopped working completely. 

Edit: This seems to be an issue when I import other currency accounts into it other than USD - it's loading very slow or not at all.
